src/marker.rs

```diff
@@ -382,6 +382,7 @@ impl<'py> Python<'py> {
     /// [`PyGILState_Ensure`]: crate::ffi::PyGILState_Ensure
     /// [`auto-initialize`]: https://pyo3.rs/main/features.html#auto-initialize
     #[inline]
+    #[deprecated(since = "0.17.0", note = "prefer Python::with_gil")]
     pub fn acquire_gil() -> GILGuard {
         GILGuard::acquire()
```

Comment:
Is it worth adding a comment here to remind us as maintainers that when we remove this API for good we can also remove the drop-order tracking from `GILGuard`?